### DISCUSSION TITLE
Add muti-factor Authentication

### DISCUSSION THREAD

**Discussion by musnows** [floxay/python-riot-auth#2] [Ideas]
```
https://github.com/techchrism/valorant-api-docs/blob/trunk/docs/Riot%20Auth/PUT%20Multi-Factor%20Authentication.md
```

Maybe muti-factor Authentication could be added?

**Comment by floxay** [comment]
Yes, it could definitely be added but I'd need to rewrite the `authorize` method to allow MFA and continue the flow after the code is entered. However I'm not sure when I'll get around to implement this.

**Reply by musnows** [reply]
> Yes, it could definitely be added but I'd need to rewrite the `authorize` method to allow MFA and continue the flow after the code is entered. However I'm not sure when I'll get around to implement this.


I‘m  coding a bot to fetch daily store. Is there any way to save the user's login information and then load it into `auth obj` after restarting? 
(except save user's account&passwd)

Now I just save `auth obj` in global value,which will clear up when shutdown the bot

**Reply by floxay** [reply]
> I‘m coding a bot to fetch daily store. Is there any way to save the user's login information and then load it into `auth obj` after restarting? (except save user's account&passwd)

The only useful thing would be the cookie jar, stored in `_cookie_jar`.
This is an aiohttp.CookieJar object which has a `save` (and `load`) method.
https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.CookieJar.save
After you loaded the saved cookies you would just call `reauthorize()` to get new tokens.

**Reply by musnows** [reply]
> > I‘m coding a bot to fetch daily store. Is there any way to save the user's login information and then load it into `auth obj` after restarting? (except save user's account&passwd)
> 
> The only useful thing would be the cookie jar, stored in `_cookie_jar`. This is an aiohttp.CookieJar object which has a `save` (and `load`) method. https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.CookieJar.save After you loaded the saved cookies you would just call `reauthorize()` to get new tokens.

I just try and it works! Thanks a lot~
